powerstore/resource_volume_group_snapshot.go

```diff
@@ -0,0 +1,264 @@
+package powerstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"log"
+	"regexp"
+	"terraform-provider-powerstore/client"
+	"terraform-provider-powerstore/models"
+)
+
+// newVGSnapshotResource returns snapshot new resource instance
+func newVGSnapshotResource() resource.Resource {
+	return &resourceVGSnapshot{}
+}
+
+type resourceVGSnapshot struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceVGSnapshot) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_volumeGroupSnapshot"
+}
+
+// Schema defines resource interface Schema method
+func (r *resourceVGSnapshot) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "Volume Group Snapshot resource",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "The unique identifier of the volume group snapshot.",
+				MarkdownDescription: "The unique identifier of the volume group snapshot.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Name of the volume group snapshot.",
+				MarkdownDescription: "Name of the volume group snapshot.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"volume_group_id": schema.StringAttribute{
+				Optional:            true,
```

Comment:
Can volume_group_id added as computed attribute? same for name as well?

Reply:
As discussed, volume group id has been made computed, but not volume group name